src/k8s/pkg/utils/k8s/restart_deployment.go

```diff
@@ -0,0 +1,31 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RestartDeployment updates the restartedAt field to trigger a rollout restart for the given Deployment.
+func RestartDeployment(ctx context.Context, client *k8sClient, name, namespace string) error {
+	if namespace == "" {
+		namespace = "default"
+	}
+	deployment, err := client.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to get deployment %s in %s: %w", name, namespace, err)
+	}
+
+	if deployment.Spec.Template.ObjectMeta.Annotations == nil {
+		deployment.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+	}
+	deployment.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
```

Comment:
Throw-away note: Is this exactly what kubectl does as well? I hope time zones etc don't become a big source of trouble here